pkg/builder/merged_entity_meta_data_builder.go

```diff
@@ -256,46 +283,63 @@ func NewMergedEntityMetadataBuilder() *MergedEntityMetadataBuilder {
 	return builder
 }
 
+// Build creates and gets the MergedEntityMetadata object
 func (builder *MergedEntityMetadataBuilder) Build() (*proto.MergedEntityMetadata, error) {
-	metadata := &proto.MergedEntityMetadata{}
-
-	// Add the internal and external property matching metadata
 	matchingMetadata, err := builder.matchingMetadataBuilder.build()
 	if err != nil {
 		return nil, err
 	}
-	metadata.MatchingMetadata = matchingMetadata
 
+	mergedEntityMetadata := &proto.MergedEntityMetadata{
+		KeepStandalone: &builder.keepStandAlone,
+		// Add the internal and external property matching metadata
+		MatchingMetadata: matchingMetadata,
+	}
+
+	// Add commodities sold list
 	if len(builder.commoditiesSold) > 0 {
-		metadata.CommoditiesSold = append(metadata.CommoditiesSold, builder.commoditiesSold...)
+		mergedEntityMetadata.CommoditiesSold = append(mergedEntityMetadata.CommoditiesSold, builder.commoditiesSold...)
+	}
+
+	// Add patched properties
+	for _, propertyBuilder := range builder.propertyBuilders {
+		mergedEntityMetadata.PatchedProperties = append(mergedEntityMetadata.PatchedProperties, propertyBuilder.build())
+	}
+
+	// Add patched fields
+	for _, fieldBuilder := range builder.fieldBuilders {
+		mergedEntityMetadata.PatchedFields = append(mergedEntityMetadata.PatchedFields, fieldBuilder.build())
 	}
 
-	if len(builder.commBoughtMetadataList) > 0 {
-		for _, commBoughtMetadata := range builder.commBoughtMetadataList {
-			commBought := &proto.MergedEntityMetadata_CommodityBoughtMetadata{
-				CommodityMetadata: commBoughtMetadata.commBoughtList,
-				ProviderType:      &commBoughtMetadata.providerType,
-			}
-			if commBoughtMetadata.replaceProvider {
-				commBought.ReplacesProvider = &commBoughtMetadata.providerToReplace
-			}
-			metadata.CommoditiesBought = append(metadata.CommoditiesBought, commBought)
-		}
+	// Add patched commodities bought
+	for _, commodityBoughtMetadataBuilder := range builder.commBoughtMetadataList {
+		mergedEntityMetadata.CommoditiesBought = append(mergedEntityMetadata.CommoditiesBought, commodityBoughtMetadataBuilder.build())
 	}
 
-	return metadata, nil
+	// Add patched commodities sold
+	for _, commoditySoldMetadataBuilder := range builder.commoditiesSoldMetadataList {
+		mergedEntityMetadata.CommoditiesSoldMetadata = append(mergedEntityMetadata.CommoditiesSoldMetadata, commoditySoldMetadataBuilder.build())
+	}
+
+	return mergedEntityMetadata, nil
 }
 
-func (builder *MergedEntityMetadataBuilder) KeepStandAlone(bool_val bool) *MergedEntityMetadataBuilder {
-	builder.metadata.KeepStandalone = &bool_val
+// KeepInTopology indicates whether the entity reported by the probe should be kept in the topology or not if no
+// stitching match is found.
+// By default (if this function is not called) an entity is not kept in the topology if no stitching match is found.
+func (builder *MergedEntityMetadataBuilder) KeepInTopology() *MergedEntityMetadataBuilder {
+	builder.keepStandAlone = true
```

Comment:
Do we still need the method here to populate the value if the value is hardcoded? I mean we can manipulate the DTO to set the value also.

Reply:
This method provides a builder method to set the `keepStandAlone` value. There is a similar method in the Java SDK:
```
    public MergedEntityMetadataBuilder keepInTopology(boolean keepInTopology) {
        builder.setKeepStandalone(keepInTopology);
        return this;
    }
```
But I realized that the default behaviour is actually to keep the proxy entity in the topology even if there is no match (i.e., the default value for keepStandAlone is true). So my current logic is wrong as there is no way to set keepStandAlone to false. I will update this method.

Reply:
Yes, but the behavior is different from Java side since the value is hardcoded here. Glad that you caught the issue.